### PR TITLE
Update project building

### PR DESCRIPTION
**Pull request by alinbalutoiu**
This PR adds the following:
- a script to setup the whole project without asking anything from the user
- moves the third party API keys in a separate file
- adds automatic detection when running inside a cloud to detect the public IP of the instance

### CODE REVIEW

File: config_vars.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+# Set to YES if running in the cloud
+RUNNING_IN_CLOUD='NO'
+
+# Trying to detect if running in a cloud...
+curl metadata.google.internal -si 2>&1>/dev/null
+if [ $? -eq 0 ]; then
+    RUNNING_IN_CLOUD='YES'
+fi
+curl http://169.254.169.254/1.0/meta-data/instance-id -s 2>&1>/dev/null
```
Comment (mircealungu) [inline config_vars.sh:16]: what's this hardcoded IP address doing here?

File: setup_all.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# Build and run the API
+docker build -t zeeguu-api-core -f docker-files/zeeguu-api-core/Dockerfile .
+docker run --net=host -d \
+    -e MICROSOFT_TRANSLATE_API_KEY="$MICROSOFT_TRANSLATE_API_KEY" \
+    -e GOOGLE_TRANSLATE_API_KEY="$GOOGLE_TRANSLATE_API_KEY" \
+    -e WORDNIK_API_KEY='$WORDNIK_API_KEY' \
+    --name=zeeguu-api-core zeeguu-api-core
+
+# Build and run the Web UI
+docker build -t zeeguu-web --build-arg ZEEGUU_API__EXTERNAL="http://$SERVER_IP:9001" -f docker-files/zeeguu-web/Dockerfile .
```
Comment (mircealungu) [inline setup_all.sh]: is it ok that the 9001 is hardcoded here?

File: setup_all.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+# Start MySQL server
+./run_mysql.sh
+
+# Build and run the API
+docker build -t zeeguu-api-core -f docker-files/zeeguu-api-core/Dockerfile .
+docker run --net=host -d \
+    -e MICROSOFT_TRANSLATE_API_KEY="$MICROSOFT_TRANSLATE_API_KEY" \
+    -e GOOGLE_TRANSLATE_API_KEY="$GOOGLE_TRANSLATE_API_KEY" \
+    -e WORDNIK_API_KEY='$WORDNIK_API_KEY' \
```
Comment (mircealungu) [inline setup_all.sh]: why is only wordnik surrounded by single quotes?

File: setup_all.sh (hunk 2 above)
Comment (mircealungu) [reply setup_all.sh]: the apache conf actually expects the api to be on port 8080 at the moment.

File: config_vars.sh (hunk 1 above)
Comment (alinbalutoiu) [reply config_vars.sh:16]: It's the cloud instance metadata URL. It's what AWS uses to pass metadata to the instances created in the cloud. https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
It helps detecting if the script is being run on an instance from inside a cloud.

File: setup_all.sh (hunk 3 above)
Comment (alinbalutoiu) [reply setup_all.sh]: Ops, that's a mistake. Single quotes in bash will not replace the variable with it's value. I will fix it.

File: setup_all.sh (hunk 2 above)
Comment (alinbalutoiu) [reply setup_all.sh]: It should be http://<server_ip>/api at the moment. The apache from Zeeguu-Web will forward requests to Zeeguu-API.

**Review by alinbalutoiu — COMMENTED** [review]
Thanks for the feedback, I will update the PR with the new changes.

**Review by mircealungu — APPROVED** [review]
cool!

**Pull request merged**